crates/node/service/src/service/validator.rs

```diff
@@ -0,0 +1,97 @@
+//! [ValidatorNodeService] trait.
+
+use crate::{DerivationActor, L2ForkchoiceState, NodeActor, service::spawn_and_wait};
+use async_trait::async_trait;
+use kona_derive::traits::{Pipeline, SignalReceiver};
+use kona_genesis::RollupConfig;
+use kona_protocol::BlockInfo;
+use std::fmt::Display;
+use tokio::sync::mpsc::{self, UnboundedSender};
+use tokio_util::sync::CancellationToken;
+
+/// The [ValidatorNodeService] trait defines the common interface for running a validator node
+/// service in the rollup node. The validator node listens to two sources of information to sync the
+/// L2 chain:
+///
+/// 1. The data availability layer, with a watcher that listens for new updates. L2 inputs (L2
+///    transaction batches + deposits) are then derived from the DA layer.
+/// 2. The L2 sequencer, which produces unsafe L2 blocks and sends them to the network over p2p
+///    gossip.
+///
+/// From these two sources, the validator node imports `unsafe` blocks from the L2 sequencer and
+/// `safe` blocks from the L2 derivation pipeline into the L2 execution layer via the Engine API.
+///
+/// Finally, a state actor listens for new L2 block import events and updates the L2 state
+/// accordingly, sending notifications to the other actors for synchronization.
+///
```

Comment:
We'll probably also want to make p2p out generic so we can sub in a noop impl if a node configs no gossip out